src/core/thread/mle.cpp

```diff
@@ -1810,6 +1810,8 @@ ThreadError Mle::HandleParentResponse(const Message &aMessage, const Ip6::Messag
     SuccessOrExit(error = Tlv::GetTlv(aMessage, Tlv::kLinkMargin, sizeof(linkMarginTlv), linkMarginTlv));
     VerifyOrExit(linkMarginTlv.IsValid(), error = kThreadError_Parse);
 
+    VerifyOrExit(threadMessageInfo != NULL, error = kThreadError_NotFound; otLogDebgMle("thread message info not found\n"));
```

Comment:
My understanding is that duplicate addresses was really the root case of your issue.  With #661, you should not be triggering this `assert()`.